python/flare/code_grafter.py

```diff
@@ -0,0 +1,858 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019 FireEye, Inc. All Rights Reserved.
+
+"""Code grafting: Static linking of code into IDBs to aid emulation."""
+
+import idc
+import idaapi
+import idautils
+import ida_xref
+import ida_bytes
+
+import mykutils
+from mykutils import phex
+from seghelper import SegPlanner
+
+import struct
+import logging
+import binascii
+
+__author__ = 'Michael Bailey'
+__copyright__ = 'Copyright (C) 2019 FireEye, Inc.'
+__license__ = 'Apache License 2.0'
+__version__ = '1.0'
+
+logger = logging.getLogger()
+logging.basicConfig(format='%(message)s', level=logging.INFO)
+
+###############################################################################
+# Constants
+###############################################################################
+METAPC = 'metapc'
+
+###############################################################################
+# Globals
+###############################################################################
+
+# Code Carving/Grafting
+
+g_seg_sig_code_grafter = 'Segment added by FLARE Code Grafter'
+g_patched_call_cmt = 'Patched for emulation, was:'
+g_cmt_pointed = '(Pointed'
+
+# Name tuple -> implementation lookup
+g_patch_pointer_width = {
+    32: idc.patch_dword,
+    64: idc.patch_qword,
+}
+
+# Per-architecture/bitness library of native function implementations. Each
+# entry is a hexadecimal ASCII string of native opcodes compiled from C and
+# specially crafted not to access globals or call other functions.
+g_fnbytes = {
+    METAPC: {
+        32: {},
+        64: {},
+    }
+}
+
+g_fnbytes[METAPC][32]['memcpy'] = (
```

Comment:
add instruction comments for memcpy & strcpy & strlen & setlocale, similar to the malloc allocator instruction breakdown in the below code.